scripts/tasks/create-trigger-group-cm.task.ts

```diff
@@ -0,0 +1,124 @@
+import { params } from './params'
+
+interface CreateTriggerGroupArgs extends BaseTaskArgs {
+    vault: BigNumber
+    type: number
+    bb: any[]
+    bs: any[]
+}
+// eg use block : 15162445 and npx hardhat create-trigger-group-cm --vault 29032 --type 1 --bb '[23200,21900,"0",true,100,200]' --bs  '[21200,21900,"0",true,100,200]' --network hardhat
+createTask<CreateTriggerGroupArgs>('create-trigger-group-cm', 'Creates an automation trigger group for a user')
+    .addParam('vault', 'The vault (cdp) ID', undefined, params.bignumber, false)
+    .addParam('type', 'The trigger group type', TriggerGroupType.CONSTANT_MULTIPLE, types.int)
+    .addParam(
+        'bb',
+        `The remaining args for the bb trigger data (i.e. '[23200,21900,"0",true,100,200]'). See 'encodeTriggerData' for more info`,
+        undefined,
+        types.json,
+        false,
+    )
+    .addParam(
+        'bs',
+        `The remaining args for the bs trigger data (i.e. '[21200,21900,"0",true,100,200]'). See 'encodeTriggerData' for more info`,
+        undefined,
+        types.json,
+        false,
+    )
+    .setAction(async (args: CreateTriggerGroupArgs, hre) => {
+        const { name: network } = hre.network
+        console.log(
+            `Network: ${network}. Using addresses from ${coalesceNetwork(args.forked || (network as Network))}\n`,
+        )
+        const hardhatUtils = new HardhatUtils(hre, args.forked)
+
+        const aggregator = await hre.ethers.getContractAt(
+            'AutomationBotAggregator',
+            hardhatUtils.addresses.AUTOMATION_BOT_AGGREGATOR,
+        )
+        const bot = await hre.ethers.getContractAt('AutomationBot', hardhatUtils.addresses.AUTOMATION_BOT)
+        let signer: Signer = hre.ethers.provider.getSigner(0)
+
+        const cdpManager = await hre.ethers.getContractAt('ManagerLike', hardhatUtils.addresses.CDP_MANAGER)
+        const proxyAddress = await cdpManager.owns(args.vault.toString())
+        const proxy = await hre.ethers.getContractAt('DsProxyLike', proxyAddress)
+        const currentProxyOwner = await proxy.owner()
+        if (currentProxyOwner.toLowerCase() !== (await signer.getAddress()).toLowerCase()) {
+            if (!isLocalNetwork(network)) {
+                throw new Error(
+                    `Signer is not an owner of the proxy. Cannot impersonate on external network. Signer: ${await signer.getAddress()}. Owner: ${currentProxyOwner}`,
+                )
+            }
+            console.log(`Impersonating proxy owner ${currentProxyOwner}...`)
+            signer = await hardhatUtils.impersonate(currentProxyOwner)
+            // Fund the owner
+            await hre.ethers.provider.getSigner(0).sendTransaction({
+                to: currentProxyOwner,
+                value: EthersBN.from(10).pow(18),
+            })
+        }
+
+        const bbTriggerData = encodeTriggerData(args.vault.toNumber(), TriggerType.CM_BASIC_BUY, ...args.bb)
```

Comment:
TriggerType.CM_BASIC_BUY can be changed back to  TriggerType.BASIC_BUY

Reply:
ok. I'm changing it and I'll send the `revert all the rest of the repo to 3 and 4` PR in a sec